thunder/core/module.py

```diff
@@ -19,16 +19,22 @@ def __init__(self, model, compiled_model_call):
         super().__init__()
         self._model = model
 
-        # We delete self.training in order for training to be used from
-        # the model itself through `__getattr__`.
         del self.training
-
         self._forward_fn = compiled_model_call
+        self._overrides = {}
 
     def get_buffer(self, name):
+        null = object()
+        p = self._overrides.get(name, null)
+        if p is not null:
+            return p
```

Comment:
why not
```python
if name in self._overrides:
    return self._overrides[name]
return self._model.get_buffer(name)
```
?

Reply:
It does one lookup more / is inherently not thread safe (not of matter here). Probably not that bad in the grand scheme of things (and compared to creating the null object).